src/macros.jl

```diff
@@ -674,11 +677,14 @@ function combine_helper(x, args...; deprecation_warning = false)
     deprecation_warning && @warn "`@based_on` is deprecated. Use `@combine` instead."
 
     # Only allow one argument when returning a Table object
-    if length(args) == 1 &&
-        !(first(args) isa QuoteNode) &&
-        !(first(args).head == :(=) || first(args).head == :kw)
+    exprs = fix_args(args...)
```

Comment:
This is complicated stuff to preserve some the 0.21.* rules about only having one `AsTable` argument. 

It's clunky and only exists to ensure tests pass. I am fine deprecating in a later PR.